lib/install/management/config.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package management
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/datastore"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+)
+
+// GuestInfoSecret downloads the VCH's vmx file and extracts the GuestInfoSecretKey
+// value from it. This function expects the datastore in the dispatcher's session
+// to be set.
+func (d *Dispatcher) GuestInfoSecret(vchName string) ([]byte, error) {
+	defer trace.End(trace.Begin(""))
+
+	if d.session.Datastore == nil {
+		return nil, fmt.Errorf("session's datastore is not set")
+	}
+
+	helper, err := datastore.NewHelper(d.ctx, d.session, d.session.Datastore, d.vmPathName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Download the VCH's vmx file
+	path := fmt.Sprintf("%s.vmx", vchName)
+	rc, err := helper.Download(d.ctx, path)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Could split this here into 2 functions, easier to unit test code below for error handling, etc.